src/Features/Core/Portable/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs

```diff
@@ -19,26 +19,22 @@ internal abstract class AbstractConvertNumericLiteralCodeRefactoringProvider : C
         {
             var document = context.Document;
             var cancellationToken = context.CancellationToken;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var numericToken = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
+                token => syntaxFacts.IsNumericLiteralExpression(token.Parent), cancellationToken).ConfigureAwait(false);
 
-            var numericToken = root.FindToken(context.Span.Start);
-            if (numericToken.ContainsDiagnostics)
+            if (numericToken == default(SyntaxToken))
             {
                 return;
             }
 
-            if (!numericToken.Span.Contains(context.Span))
```

Comment:
I think that'd be unexpected if we don't offer the fix when the selection matches the token because this refactoring operates on a single token. On the other hand, when we offer a refactoring based on parents of a node e.g. "add argument name", then the no-selection approach would be preferable imo.